app/command_groupmsg.go

```diff
@@ -0,0 +1,148 @@
+		AutoCompleteDesc: T("api.command_groupmsg.desc"),
+		AutoCompleteHint: T("api.command_groupmsg.hint"),
+		DisplayName:      T("api.command_groupmsg.name"),
+	}
+}
+
+func (me *groupmsgProvider) DoCommand(a *App, args *model.CommandArgs, message string) *model.CommandResponse {
+	parsedMessage := ""
+	targetUsers := map[string]*model.User{}
+	targetUsersSlice := []string{args.UserId}
+	invalidUsernames := []string{}
+
+	userExpr := "@?[^\\s,]+"
+	sepExpr := "\\s*,\\s*"
+	textExpr := "\\s*([^,].*)?"
+	messageExpr := fmt.Sprintf("^\\s*((?:%s%s)*%s)%s?$", userExpr, sepExpr, userExpr, textExpr)
+	messageRegex := regexp.MustCompile(messageExpr)
+	if !messageRegex.MatchString(message) {
+		return &model.CommandResponse{Text: args.T("api.command_groupmsg.invalid_arguments.app_error"), ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL}
+	}
+
+	matches := messageRegex.FindStringSubmatch(message)
+
+	for _, username := range strings.Split(matches[1], ",") {
+		username = strings.TrimSpace(username)
+		username = strings.TrimPrefix(username, "@")
+		if result := <-a.Srv.Store.User().GetByUsername(username); result.Err != nil {
+			invalidUsernames = append(invalidUsernames, username)
+		} else {
+			targetUser := result.Data.(*model.User)
+			_, exists := targetUsers[targetUser.Id]
+			if !exists && targetUser.Id != args.UserId {
+				targetUsers[targetUser.Id] = targetUser
+				targetUsersSlice = append(targetUsersSlice, targetUser.Id)
+			}
+		}
+	}
+
+	if len(invalidUsernames) > 0 {
+		invalidUsersString := map[string]interface{}{
+			"Users": "@" + strings.Join(invalidUsernames, ", @"),
+		}
+		return &model.CommandResponse{
+			Text:         args.T("api.command_groupmsg.invalid_user.app_error", len(invalidUsernames), invalidUsersString),
+			ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+		}
+	}
+
+	if len(matches) > 2 {
+		parsedMessage = matches[2]
+	}
+
+	if len(targetUsersSlice) == 2 {
+		return GetCommandProvider("msg").DoCommand(a, args, fmt.Sprintf("%s %s", targetUsers[targetUsersSlice[1]].Username, parsedMessage))
+	}
+
+	if len(targetUsersSlice) < model.CHANNEL_GROUP_MIN_USERS {
+		minUsers := map[string]interface{}{
+			"MinUsers": model.CHANNEL_GROUP_MIN_USERS - 1,
+		}
+		return &model.CommandResponse{
+			Text:         args.T("api.command_groupmsg.min_users.app_error", minUsers),
+			ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+		}
+	}
+
+	if len(targetUsersSlice) > model.CHANNEL_GROUP_MAX_USERS {
+		maxUsers := map[string]interface{}{
+			"MaxUsers": model.CHANNEL_GROUP_MAX_USERS - 1,
+		}
+		return &model.CommandResponse{
+			Text:         args.T("api.command_groupmsg.max_users.app_error", maxUsers),
+			ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+		}
+	}
+
+	// Find the channel based on this user
+	channelName := model.GetGroupNameFromUserIds(targetUsersSlice)
+
+	targetChannelId := ""
+	if channel := <-a.Srv.Store.Channel().GetByName(args.TeamId, channelName, true); channel.Err != nil {
```

Comment:
I'm pretty sure `(*App).CreateGroupChannel` will return the existing channel if there already is one. If so, you can simplify this a lot.

Reply:
Yes, you are right, I changed it and simplified the code. :)

Reply:
I removed the regexp and change with the code and some unit tests to ensure that is working correctly :)

Reply:
And I fixed the "4 spaces" style, I have to configure correctly my vim to reformat it correctly.